### ISSUE TITLE
Process lingers and uses quite a bit of resources

### ISSUE THREAD

**Issue by Garanas** [FAForever/downlords-faf-client#3016]
**Describe the bug**

The process lingers after exiting for quite a while and consumes quite a bit of resources:

![image](https://github.com/FAForever/downlords-faf-client/assets/15778155/62829019-c1f6-4c30-b930-275bd729ec9a)

**To Reproduce**

Start the application, let it stay in for a few hours and exit it.

**OS**
Windows

**Comment by Sheikah45** [comment]
The UI closes immediately but the process has to clean up after self and do things like write the preferences to disk so this is as expected and not something to be fixed.

I don't understand what you are looking for as a solution here or why this is a problem.

**Comment by valignatev** [comment]
this is a huge problem, because faf-client.exe can hang while taking 40% of 10 core cpu and being at very high power usage for a long time. I'm talking lilke 10+ minutes. It's definitely not just writing game prefs to disk. It also means that you can't restart the client without killing this process through task manager first. It's like this for 2+ months already. IT also starts eating more and more CPU and more and more power even before you close the UI. At some point, the client becomes unusable

**Comment by BlackYps** [comment]
Just to clarify, the process exists for 10+ minutes after you closed the ui?
Which version of windows are you using?

**Comment by Sheikah45** [comment]
Do you have a client.log from when it takes 10+ minutes?

**Comment by Garanas** [comment]
> The UI closes immediately but the process has to clean up after self and do things like write the preferences to disk so this is as expected and not something to be fixed.
> 
> I don't understand what you are looking for as a solution here or why this is a problem.

I understand that it needs to finish up a task or two, but using so much system resources for such a long duration (several minutes) feels like a bit much.

I tried to find a log for you, but it didn't write it after killing the process 😄 . The logs are quite large (10MBs+ worth of disk storage)

**Comment by valignatev** [comment]
I'm on windows 10. The process sometimes closes after a minute or so, and some times just hangs in there until I kill it. Not sure which logs would be useful, but here's some big one. It's from yesterday, and yesterday it happened multiple times. faf client was already idling at 40% cpu and high power usage even before I killed it, so maybe there's something in the log
[client.log.2023-07-23.0.log](https://github.com/FAForever/downlords-faf-client/files/12144994/client.log.2023-07-23.0.log)

**Comment by SHOTbyGUN** [comment]
Linux: 400% CPU usage 2gig ram usage. Longer the client has been running, higher the probability that it does not close right. Bad behavior starts when pressing the closing X and CPU usage shoots up. Almost as if some threads gets into infinite loop, where there is no work but exit condition is not met. If I start and close the FAF instantly, it closes as expected.

Update: Happened today again, let it run and it finally closed after ~30 seconds. I always terminated it much sooner. When the client closed, computer time was 15:47

Last lines on client.log:

```
2023-07-25T15:42:30.044+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Fetching image from https://content.faforever.com/maps/previews/small/foxtrot_runway.v0002.png
2023-07-25T15:42:50.051+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Fetching image from https://content.faforever.com/maps/previews/small/x1ca_coop_001.v0028.png
2023-07-25T15:42:50.241+03:00  WARN 23596 --- [ForkJoinPool.commonPool-worker-5] com.faforever.client.fx.JavaFxUtil       : Could not read image from https://content.faforever.com/maps/previews/small/x1ca_coop_001.v0028.png for %USER_PROFILE%/.faforever/cache/maps/small/x1ca_coop_001.v0028.png
2023-07-25T15:42:52.063+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/gap_of_rohan_forever.v0007.png
2023-07-25T15:42:52.071+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Fetching image from https://content.faforever.com/maps/previews/small/scca_coop_r04.v0019.png
2023-07-25T15:42:52.200+03:00  WARN 23596 --- [ForkJoinPool.commonPool-worker-5] com.faforever.client.fx.JavaFxUtil       : Could not read image from https://content.faforever.com/maps/previews/small/scca_coop_r04.v0019.png for %USER_PROFILE%/.faforever/cache/maps/small/scca_coop_r04.v0019.png
2023-07-25T15:43:45.409+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/the_ganges_chasma.v0002.png
2023-07-25T15:44:57.594+03:00  INFO 23596 --- [taskScheduler-1] c.f.client.config.PreferencesConfig      : Writing preferences file `%USER_PROFILE%/.faforever/client.prefs`
2023-07-25T15:45:00.067+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/survival_5thdimension_v9.v0001.png
2023-07-25T15:45:13.066+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Fetching image from https://content.faforever.com/maps/previews/small/x1ca_coop_005.v0022.png
2023-07-25T15:45:13.066+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Fetching image from https://content.faforever.com/maps/previews/small/x1ca_coop_002.v0023.png
2023-07-25T15:45:13.183+03:00  WARN 23596 --- [ForkJoinPool.commonPool-worker-5] com.faforever.client.fx.JavaFxUtil       : Could not read image from https://content.faforever.com/maps/previews/small/x1ca_coop_005.v0022.png for %USER_PROFILE%/.faforever/cache/maps/small/x1ca_coop_005.v0022.png
2023-07-25T15:45:13.193+03:00  WARN 23596 --- [ForkJoinPool.commonPool-worker-5] com.faforever.client.fx.JavaFxUtil       : Could not read image from https://content.faforever.com/maps/previews/small/x1ca_coop_002.v0023.png for %USER_PROFILE%/.faforever/cache/maps/small/x1ca_coop_002.v0023.png
2023-07-25T15:45:15.071+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/astro_crater_battles_4x4_rich_huge.v0004.png
2023-07-25T15:46:53.562+03:00  INFO 23596 --- [JavaFX Application Thread] c.faforever.client.steam.SteamService    : Stopping the Steam API
2023-07-25T15:46:53.568+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.steam.SteamService    : Steam API stopped
2023-07-25T15:46:53.568+03:00  INFO 23596 --- [JavaFX Application Thread] c.f.client.fa.relay.ice.IceAdapterImpl   : Ignoring call to ICE adapter as we are not connected: quit([])
2023-07-25T15:46:53.570+03:00  INFO 23596 --- [JavaFX Application Thread] c.f.c.map.generator.MapGeneratorService  : Deleting generated maps
2023-07-25T15:46:53.574+03:00  INFO 23596 --- [Thread-13] com.faforever.client.map.MapService      : Watcher terminated (null)
2023-07-25T15:46:53.574+03:00 DEBUG 23596 --- [Thread-14] com.faforever.client.mod.ModService      : Watcher terminated (null)
2023-07-25T15:46:53.575+03:00  INFO 23596 --- [pool-6-thread-1] com.faforever.client.theme.UiService     : Watcher service terminated
2023-07-25T15:46:53.575+03:00  INFO 23596 --- [JavaFX Application Thread] c.faforever.client.config.AsyncConfig    : Shutting down ExecutorService 'taskExecutor'
2023-07-25T15:46:53.576+03:00  INFO 23596 --- [JavaFX Application Thread] c.f.client.remote.FafServerAccessor      : Closing lobby server connection
2023-07-25T15:46:53.576+03:00  INFO 23596 --- [reactor-tcp-epoll-1] c.faforever.commons.lobby.FafLobbyApi    : Inbound channel closed
2023-07-25T15:46:53.576+03:00  INFO 23596 --- [reactor-tcp-epoll-1] c.faforever.commons.lobby.FafLobbyApi    : Outbound channel cancelled
2023-07-25T15:46:53.577+03:00  INFO 23596 --- [reactor-tcp-epoll-1] c.faforever.commons.lobby.FafLobbyApi    : Disconnected from server
2023-07-25T15:46:55.612+03:00  INFO 23596 --- [JavaFX Application Thread] c.f.client.config.PreferencesConfig      : Writing preferences file `%USER_PROFILE%/.faforever/client.prefs`
2023-07-25T15:46:55.630+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.632+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.633+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.634+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.635+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.635+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.636+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.638+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Fetching image from https://content.faforever.com/maps/previews/small/x1ca_coop_002.v0023.png
2023-07-25T15:46:55.639+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Fetching image from https://content.faforever.com/maps/previews/small/x1ca_coop_005.v0022.png
2023-07-25T15:46:55.640+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.640+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.642+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.643+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.645+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.645+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.647+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.647+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.649+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.649+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.651+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.651+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.653+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.653+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.655+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.656+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.657+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.658+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.659+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.660+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.661+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.662+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.663+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.664+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.665+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.666+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.667+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.667+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.669+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Fetching image from https://content.faforever.com/maps/previews/small/x1ca_coop_002.v0023.png
2023-07-25T15:46:55.669+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Fetching image from https://content.faforever.com/maps/previews/small/x1ca_coop_005.v0022.png
2023-07-25T15:46:55.670+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.671+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.672+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.673+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.674+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.675+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.676+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.677+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.681+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/setonnn.v0003.png
2023-07-25T15:46:55.682+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/setonnn.v0003.png
2023-07-25T15:46:55.682+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/08_you_shall_not_pass.v0002.png
2023-07-25T15:46:55.683+03:00 DEBUG 23596 --- [JavaFX Application Thread] c.faforever.client.remote.AssetService   : Using cached image: %USER_PROFILE%/.faforever/cache/maps/small/setonnn.v0003.png
2023-07-25T15:46:55.728+03:00  WARN 23596 --- [JavaFX Application Thread] c.f.c.g.GamesTilesContainerController    : Tried to remove game tile that did not exist.
```

**Comment by valignatev** [comment]
Same thing. If it runs for more than an hour it's pretty much impossible for faf-client process to reliably exit. Performance also degrades the longer it stays open, so I suspect there might be memory leaks as well

**Comment by ChessBerry** [comment]
Same for me.

As a side effect the process staying open for a long time after the client closes means that you can't quickly restart it w/o using the task manager to kill it manually. Otherwise pressing the faf client exe just does nothing.

Possibly related to this and/or #3012: If I close the client after it being open for a couple of hours, it happens semi-reliably that, like 5 minutes after closing it, I hear the distinct "message received" sound from the client, after which the process is also gone from the task manager.
This has happened like half a dozen of times over the last weeks, so I'm certain I didn't imagine it, but the correlation might of course be spurious.

**Comment by Sheikah45** [comment]
There is no errors in the logs that would cause a long open time. Does the client use high resources during runtime?

**Comment by valignatev** [comment]
Yes for me, the longer it runs the more it eats. Also, the performance degrades drastically over time.

**Comment by Sheikah45** [comment]
Can you give more information on your system as I am unable to replicate it as when I run it locally for stress tests over a week it remains at a stabe resource usage. Can you also describe what actions you normally take with like the interactions.

**Comment by valignatev** [comment]
faf client runs on windows 10. It's installed on SSD. CPU is intel core i9-9900k, gpu is radeon 7900xtx, although it also happened on my previous gpu, which were nvidia 1070gtx.
I would normally open the client, look at the lobbies, click some tabs around to watch replays and all that. Sometimes join a lobby and play, then I'd leave the client open and go on with my day, while occasionally read on it to read aeolus and watch more replays and stuff like that. Pretty normal usage. Sometimes I'd open a local folders through the menu, to run faf offline.
Sometimes I also put my pc to sleep, and then wake it up. Faf client would resume normally, but closing it would also leave faf-client.exe in shambles.

Overall, it seems like the more actions you do with the client, the faster it slows down.

When I just open the client, after it logged me in, I'd see this situation:
![image](https://github.com/FAForever/downlords-faf-client/assets/7038954/bf517e26-b9fd-455b-af72-7ff38d82aa07)

After clicking around, just like randomly between tabs, maybe loading my replays list, RAM usage gradually gets up but doesn't seem to ever go down.

**Comment by Sheikah45** [comment]
What build of windows 10 are you using?

**Comment by valignatev** [comment]
Edition	Windows 10 Pro
Version	21H2
Installed on	‎04.‎06.‎2020
OS build	19044.3086
Experience	Windows Feature Experience Pack 1000.19041.1000.0

**Comment by micheljung** [comment]
A (or three) thread dump(s) would help to diagnose the issue. You should be able to get them using [jattach](https://github.com/jattach/jattach).

**Comment by valignatev** [comment]
Do I just attach with jattach during the normal flow and then close the client? I assume I attach to faf-client.exe, right?

**Comment by micheljung** [comment]
@valignatev sorry for the late reply.

1. Close the client
2. Check the task manager for the PID (process ID)
3. When the process is still running, run `jattach <pid> threaddump > <file>` to write the output into `<file>`. For instance:
   ```
   jattach 2375 threaddump > C:\users\valignatev\Desktop\stackdump.txt
   ```

**Comment by valignatev** [comment]
Hey no problem. But actually since one of the recent updates I don't have this issue as bad anymore. Now, faf-client.exe would just chill for 5-ish seconds after the client is closed and then exit successfully. I also don't have the accumulated lag anymore, so don't have to restart the client all the time. But I'll do the threaddump if this ever comes back

**Comment by Sheikah45** [comment]
I think I found the issue. The Chat service tried to reconnect after the connection was closed always XD

**Comment by valignatev** [comment]
Actually, it makes sense! Right after i posted my previous message I had a situation where chat disconnected in the client and I've tried to restart the whole program to reconnect again (because trying to reconnect to the chat from the client never worked for me for some reason). And faf-client.exe stuck again for like a solid minute. But then I restarted the client again in hopes that the bug came back and I can threaddump it, but it was fine haha
